src/components/users/Login/Login.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import Input from '../../common/Input/Input';
+import Button from '../../common/Button/Button';
+import * as actions from '../../../actions';
+
+export class Login extends Component {
+  state = {
+    loginForm: {
+      email: {
+        elementType: 'input',
+        elementConfig: {
+          type: 'email',
+          placeholder: 'Enter your email'
+        },
+        value: '',
+        labelValue: 'E-Mail Address',
+        validation: {
+          required: true,
+          isEmail: true
+        },
+        valid: false,
+        touched: false,
+      },
+      password: {
+        elementType: 'input',
+        elementConfig: {
+          type: 'password',
+          placeholder: 'Enter your password'
+        },
+        value: '',
+        labelValue: 'password',
+        validation: {
+          require: true,
+          minimum: 3
+        },
+        valid: false,
+        touched: false,
+      }
+    },
+    isFormValid: false
+  }
+
+  checkValidity = (inputValue, rules) => {
+    let isValid = true;
+
+    if (rules.required) {
+      isValid = inputValue.trim() !== '' && isValid;
+    }
+
+    if (rules.minimum) {
+      isValid = inputValue.length >= rules.minimum && isValid;
+    }
+
+    if (rules.isEmail) {
+      const pattern = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
+      isValid = pattern.test(inputValue) && isValid;
+    }
+    return isValid;
+  }
+
+  submitHandler = (e) => {
+    e.preventDefault();
+    this.props.onLogin(this.state.loginForm.email.value, this.state.loginForm.password.value);
+  }
+
+  inputChangedHandler = (event, formInputName) => {
+    const updatedLoginForm = { ...this.state.loginForm };
+    const updatedElement = { ...updatedLoginForm[formInputName] };
+    updatedElement.value = event.target.value;
+    updatedElement.valid = this.checkValidity(
+      event.target.value,
+      this.state.loginForm[formInputName].validation
+    );
+    updatedElement.touched = true;
+    updatedLoginForm[formInputName] = updatedElement;
+
+    let isFormValid = true;
+    const formInputNames = Object.keys(updatedLoginForm);
+    for (let i = 0; i < formInputNames.length; i += 1) {
```

Comment:
You can use a `map` to loop through a list. Follow this link to see how the map  function is used in react https://reactjs.org/docs/lists-and-keys.html